torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -529,10 +539,17 @@ class ShapePropagator {
         return;
       }
       case aten::_unwrap_optional: {
+	// if we have None as input, we need to leave the output alone
         auto input_ivalue = toIValue(node->input());
         if (input_ivalue && input_ivalue->isNone()) {
           return;
         }
+	if(auto ot = node->input()->type()->cast<OptionalType>()) {
+	  node->output()->setType(ot->getElementType());
+	} else {
+	  node->output()->setType(node->input()->type());
+	}
```

Comment:
Why did we add this? How is `_unwrap_optional` different from `unchecked_unwrap_optional`? This code looks really weird, because we're trying to see if we can infer statically that this thing gets a `None` as an argument (which it never should??), and otherwise we assume it doesn't?

Reply:
Ok. I think we should revert this and probably also abandon the newer patch.

Reply:
_unwrap_optional is user code. it throws an error if the input is None and returns the input. unchecked_unwrap_optional has been inserted by the compiler when we can statically reason that the input will never be none (and so doesn't check if the input is None)